langchain/src/vectorstores/prisma.ts

```diff
@@ -410,9 +412,29 @@ export class PrismaVectorStore<
         Object.entries(ops).map(([opName, value]) => {
           // column name, operators cannot be parametrised
           // these fields are thus not escaped by Prisma and can be dangerous if user input is used
+          const opNameKey = opName as keyof typeof OpMap;
           const colRaw = this.Prisma.raw(`"${key}"`);
-          const opRaw = this.Prisma.raw(OpMap[opName as keyof typeof OpMap]);
-          return this.Prisma.sql`${colRaw} ${opRaw} ${value}`;
+          const opRaw = this.Prisma.raw(OpMap[opNameKey]);
+
+          switch (OpMap[opNameKey]) {
+            case OpMap.in: {
+              if (
+                !Array.isArray(value) ||
+                !value.every((v) => typeof v === "string")
```

Comment:
The proposed implementation only supports string. The fix can be generalized to support other types.

```js
buildSqlFilterStr(filter?: any) {
  if (filter == null) return null;
  // @ts-ignore
  const sql = this.Prisma.join(
    Object.entries(filter).flatMap(([key, ops]) =>
      // @ts-ignore
      Object.entries(ops).map(([opName, value]) => {
        // Patch 1 starts here (support enums)
        const cast = typeof value === 'string' ? '::text' : '';
        // Patch 1 ends here
        // @ts-ignore
        const colRaw = this.Prisma.raw(`"${key}"${cast}`);
        // @ts-ignore
        const opRaw = this.Prisma.raw(OpMap[opName as keyof typeof OpMap]);
        // Patch 2 starts here (supports all types)
        if (opName === 'in') {
          // @ts-ignore
          if (value.length === 0) {
            // @ts-ignore
            value.push(0);
          }
          // @ts-ignore
          return this.Prisma.sql`${colRaw} ${opRaw} (${this.Prisma.join(value)})`;
        }
        // Patch 2 ends here
        // @ts-ignore
        return this.Prisma.sql`${colRaw} ${opRaw} ${value}`;
      })
    ),
    ' AND ',
    ' WHERE '
  );
  return sql;
};
```